### PR TITLE
remove TODOs from http package and prober

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#### What type of PR is this?

/kind cleanup
/area kubelet
/sig node

#### What this PR does / why we need it:

This is a clean up of old workaround with three http connection pools that was fixed in https://github.com/golang/go/issues/22091. Also removed a few old TODOs.

#### Special notes for your reviewer:

I was looking at something else in this code and found that these old TODOs will unlikely be fixed, and the workaround with three http pools is unnatural and not needed any longer. Just a simple clean up.

```release-note
NONE
```

### CODE REVIEW

**Review by tallclair — COMMENTED** [review]
/lgtm
/approve

**Comment by ehashman** [comment]
/priority backlog
/triage accepted
/lgtm cancel

need to fix the linting error

File: pkg/kubelet/prober/prober.go [hunk 1]
```diff
@@ -49,16 +49,11 @@ const maxProbeRetries = 3
 
 // Prober helps to check the liveness/readiness/startup of a container.
 type prober struct {
-	exec execprobe.Prober
-	// probe types needs different httpprobe instances so they don't
-	// share a connection pool which can cause collisions to the
-	// same host:port and transient failures. See #49740.
```
Comment (ehashman) [inline pkg/kubelet/prober/prober.go:55]: n.b. https://github.com/kubernetes/kubernetes/issues/49740 was a workaround, fixed upstream in golang 1.10+ with https://github.com/golang/go/issues/22091

Should be safe to collapse these.

**Review by ehashman — COMMENTED** [review]
LGTM other than the linter failure

**Comment by matthyx** [comment]
I guess we can continue with this one...
/lgtm

**Comment by matthyx** [comment]
Ping @tallclair can you approve?

**Comment by tallclair** [comment]
no, looks like you need a sig-node-approver (I already approved, but it only counts for pkg/kubelet/prober)

**Comment by aojea** [comment]
/lgtm

nice, this divide by 3 the number of clients

**Review by mrunalp — APPROVED** [review]

**Pull request merged**